code42/py42
# Feature/integ 1850 api client initializer

```
NOTE: This branch does not target `main`.

Api-clients functionality should only be merged to main when:
 -  all external dependencies have deployed
 -  all relevent py42 PR merged into the `feature/api-clients` branch.
```

### Description of Change ###

Adds support for API clients to py42 via a new constructor for the main `SDKClient` object, `py42.sdk.from_api_client()`.

### Testing Procedure ###

- create an API client using an account in Code42's core-int environment. A link to the console page where this can be done is in the internal ticket description.

- instantiate py42 via the new method and execute a few API calls as normal. The calls should work as expected (except for a small number of services that still lack API client support).

```python
import py42.sdk

sdk = py42.sdk.from_api_client("something.code42.com", "myclientid", "mysecret")

# some examples of API calls. We really only need _one_ of these to work to prove
# that we're doing everything right, otherwise that would indicate a problem on the service side.

print(list(sdk.auditlogs.get_all()))
print(list(sdk.devices.get_all()))
print(list(sdk.users.get_all()))
print(list(sdk.trustedactivities.get_all()))
print(list(sdk.cases.get_all()))
``` 

Additionally,  using an invalid client id / secret should result in Unauthorized.

### PR Checklist ###
Did you remember to do the below?
